File: sdk/python/examples/basic_usage.py
"""
Example: Basic usage of LX-Proxy Python SDK
"""

import asyncio
from lxproxy import LXProxyClient, AuthenticationError, APIError


async def example_basic_usage():
    """Basic async usage example"""
    print("=== Basic Usage Example ===\n")
    
    async with LXProxyClient("http://localhost:8080") as client:
        try:
            # Login
            print("Logging in...")
            result = await client.async_login("admin", "admin123")
            print(f"✅ Logged in as: {result['user']['username']}")
            print(f"   Role: {result['user']['role']}\n")
            
            # Get current user
            print("Getting current user info...")
            user = await client.async_get_current_user()
            print(f"   User ID: {user.id}")
            print(f"   Username: {user.username}\n")
            
            # List users
            print("Listing users...")
            users = await client.async_list_users()
            print(f"   Total users: {len(users)}")
            for u in users:
                print(f"   - {u.username} ({u.role})")
            print()
            
            # Get system stats
            print("Getting system statistics...")
            stats = await client.async_get_stats()
            print(f"   Total users: {stats.total_users}")
            print(f"   Total inbounds: {stats.total_inbounds}")
            print(f"   Enabled inbounds: {stats.enabled_inbounds}")
            print(f"   Traffic used: {stats.total_traffic_used:,} bytes\n")
            
            # Get system status
            print("Getting system status...")
            status = await client.async_get_system_status()
            print(f"   CPU Usage: {status.cpu_usage:.1f}%")
            print(f"   Memory: {status.memory_used:,} / {status.memory_total:,} bytes")
            print(f"   Xray Running: {status.xray_running}")
            print(f"   Active Connections: {status.connections}\n")
            
            # List inbounds
            print("Listing inbounds...")
            inbounds = await client.async_list_inbounds()
            print(f"   Total inbounds: {len(inbounds)}")
            for inbound in inbounds:
                status_icon = "✅" if inbound.enable else "❌"
                print(f"   {status_icon} {inbound.tag} (Port: {inbound.port}, Protocol: {inbound.protocol})")
            print()
            
        except AuthenticationError as e:
            print(f"❌ Authentication failed: {e}")
        except APIError as e:
            print(f"❌ API error: {e} (Status: {e.status_code})")
        except Exception as e:
            print(f"❌ Unexpected error: {e}")


async def example_user_management():
    """User management example"""
    print("\n=== User Management Example ===\n")
    
    async with LXProxyClient("http://localhost:8080") as client:
        await client.async_login("admin", "admin123")
        
        # Create a new user
        print("Creating new user...")
        try:
            new_user = await client.async_create_user(
                username="demo_user",
                password="demo_password_123",
                role="user"
            )
            print(f"✅ Created user: {new_user.username} (ID: {new_user.id})")
            
            # Get user
            print("\nGetting user details...")
            user = await client.async_get_user(new_user.id)
            print(f"   Username: {user.username}")
            print(f"   Role: {user.role}")
            print(f"   Created: {user.created_at}")
            
            # Update user
            print("\nUpdating user role...")
            updated = await client.async_update_user(new_user.id, role="admin")
            print(f"✅ Updated role to: {updated.role}")
            
            # Delete user
            print("\nDeleting user...")
            await client.async_delete_user(new_user.id)
            print(f"✅ Deleted user: {new_user.username}")
            
        except APIError as e:
            print(f"❌ Error: {e} (Status: {e.status_code})")


async def example_inbound_management():
    """Inbound management example"""
    print("\n=== Inbound Management Example ===\n")
    
    async with LXProxyClient("http://localhost:8080") as client:
        await client.async_login("admin", "admin123")
        
        # Create a new inbound
        print("Creating new inbound...")
        try:
            inbound = await client.async_create_inbound(
                tag="vmess-demo",
                protocol="vmess",
                port=10086,
                traffic_limit=10737418240,  # 10GB
            )
            print(f"✅ Created inbound: {inbound.tag}")
            print(f"   ID: {inbound.id}")
            print(f"   Port: {inbound.port}")
            print(f"   Protocol: {inbound.protocol}")
            print(f"   Traffic Limit: {inbound.traffic_limit:,} bytes")
            
            # Get inbound
            print("\nGetting inbound details...")
            retrieved = await client.async_get_inbound(inbound.id)
            print(f"   Tag: {retrieved.tag}")
            print(f"   Enabled: {retrieved.enable}")
            
            # Update inbound
            print("\nUpdating inbound...")
            updated = await client.async_update_inbound(
                inbound.id,
                traffic_limit=21474836480  # 20GB
            )
            print(f"✅ Updated traffic limit to: {updated.traffic_limit:,} bytes")
            
            # Delete inbound
            print("\nDeleting inbound...")
            await client.async_delete_inbound(inbound.id)
            print(f"✅ Deleted inbound: {inbound.tag}")
            
        except APIError as e:
            print(f"❌ Error: {e} (Status: {e.status_code})")


def example_sync_usage():
    """Synchronous usage example"""
    print("\n=== Sync Usage Example ===\n")
    
    client = LXProxyClient("http://localhost:8080")
    
    try:
        # Login
        print("Logging in (sync)...")
        result = client.login("admin", "admin123")
        print(f"✅ Logged in as: {result['user']['username']}")
        
        # List users
        print("\nListing users (sync)...")
        users = client.list_users()
        print(f"   Total users: {len(users)}")
        
        # Get stats
        print("\nGetting stats (sync)...")
        stats = client.get_stats()
        print(f"   Total inbounds: {stats.total_inbounds}")
        
    except Exception as e:
        print(f"❌ Error: {e}")
    finally:
        client.logout()
        print("\n✅ Logged out")


async def main():
    """Run all examples"""
    print("LX-Proxy Python SDK - Examples\n")
    print("=" * 50)
    
    # Run async examples
    await example_basic_usage()
    await example_user_management()
    await example_inbound_management()
    
    # Run sync example
    example_sync_usage()
    
    print("\n" + "=" * 50)
    print("✅ All examples completed!\n")


if __name__ == "__main__":
    asyncio.run(main())

File: sdk/nodejs/examples/basic-usage.ts
/**
 * Example: Basic usage of LX-Proxy Node.js SDK
 */

import { LXProxyClient, AuthenticationError, APIError } from '../src/index';

async function exampleBasicUsage() {
  console.log('=== Basic Usage Example ===\n');

  // Create client
  const client = new LXProxyClient({
    baseURL: 'http://localhost:8080',
    timeout: 30000,
  });

  try {
    // Login
    console.log('Logging in...');
    const loginResult = await client.login('admin', 'admin123');
    console.log(`✅ Logged in as: ${loginResult.user.username}`);
    console.log(`   Role: ${loginResult.user.role}\n`);

    // Get current user
    console.log('Getting current user info...');
    const user = await client.getCurrentUser();
    console.log(`   User ID: ${user.id}`);
    console.log(`   Username: ${user.username}\n`);

    // List users
    console.log('Listing users...');
    const users = await client.listUsers();
    console.log(`   Total users: ${users.length}`);
    for (const u of users) {
      console.log(`   - ${u.username} (${u.role})`);
    }
    console.log();

    // Get system stats
    console.log('Getting system statistics...');
    const stats = await client.getStats();
    console.log(`   Total users: ${stats.total_users}`);
    console.log(`   Total inbounds: ${stats.total_inbounds}`);
    console.log(`   Enabled inbounds: ${stats.enabled_inbounds}`);
    console.log(`   Traffic used: ${stats.total_traffic_used.toLocaleString()} bytes\n`);

    // Get system status
    console.log('Getting system status...');
    const status = await client.getSystemStatus();
    console.log(`   CPU Usage: ${status.cpu_usage.toFixed(1)}%`);
    console.log(`   Memory: ${status.memory_used.toLocaleString()} / ${status.memory_total.toLocaleString()} bytes`);
    console.log(`   Xray Running: ${status.xray_running}`);
    console.log(`   Active Connections: ${status.connections}\n`);

    // List inbounds
    console.log('Listing inbounds...');
    const inbounds = await client.listInbounds();
    console.log(`   Total inbounds: ${inbounds.length}`);
    for (const inbound of inbounds) {
      const statusIcon = inbound.enable ? '✅' : '❌';
      console.log(`   ${statusIcon} ${inbound.tag} (Port: ${inbound.port}, Protocol: ${inbound.protocol})`);
    }
    console.log();

  } catch (error) {
    if (error instanceof AuthenticationError) {
      console.error(`❌ Authentication failed: ${error.message}`);
    } else if (error instanceof APIError) {
      console.error(`❌ API error: ${error.message} (Status: ${error.statusCode})`);
    } else {
      console.error(`❌ Unexpected error: ${error}`);
    }
  } finally {
    // Logout
    await client.logout();
    console.log('✅ Logged out\n');
  }
}

async function exampleUserManagement() {
  console.log('\n=== User Management Example ===\n');

  const client = new LXProxyClient('http://localhost:8080');

  try {
    await client.login('admin', 'admin123');

    // Create a new user
    console.log('Creating new user...');
    const newUser = await client.createUser({
      username: 'demo_user',
      password: 'demo_password_123',
      role: 'user',
    });
    console.log(`✅ Created user: ${newUser.username} (ID: ${newUser.id})`);

    // Get user
    console.log('\nGetting user details...');
    const user = await client.getUser(newUser.id);
    console.log(`   Username: ${user.username}`);
    console.log(`   Role: ${user.role}`);
    console.log(`   Created: ${user.created_at}`);

    // Update user
    console.log('\nUpdating user role...');
    const updated = await client.updateUser(newUser.id, { role: 'admin' });
    console.log(`✅ Updated role to: ${updated.role}`);

    // Delete user
    console.log('\nDeleting user...');
    await client.deleteUser(newUser.id);
    console.log(`✅ Deleted user: ${newUser.username}`);

  } catch (error) {
    if (error instanceof APIError) {
      console.error(`❌ Error: ${error.message} (Status: ${error.statusCode})`);
    } else {
      console.error(`❌ Error: ${error}`);
    }
  }
}

async function exampleInboundManagement() {
  console.log('\n=== Inbound Management Example ===\n');

  const client = new LXProxyClient('http://localhost:8080');

  try {
    await client.login('admin', 'admin123');

    // Create a new inbound
    console.log('Creating new inbound...');
    const inbound = await client.createInbound({
      tag: 'vmess-demo',
      protocol: 'vmess',
      port: 10086,
      traffic_limit: 10737418240, // 10GB
    });
    console.log(`✅ Created inbound: ${inbound.tag}`);
    console.log(`   ID: ${inbound.id}`);
    console.log(`   Port: ${inbound.port}`);
    console.log(`   Protocol: ${inbound.protocol}`);
    console.log(`   Traffic Limit: ${inbound.traffic_limit?.toLocaleString()} bytes`);

    // Get inbound
    console.log('\nGetting inbound details...');
    const retrieved = await client.getInbound(inbound.id);
    console.log(`   Tag: ${retrieved.tag}`);
    console.log(`   Enabled: ${retrieved.enable}`);

    // Update inbound
    console.log('\nUpdating inbound...');
    const updated = await client.updateInbound(inbound.id, {
      traffic_limit: 21474836480, // 20GB
    });
    console.log(`✅ Updated traffic limit to: ${updated.traffic_limit?.toLocaleString()} bytes`);

    // Delete inbound
    console.log('\nDeleting inbound...');
    await client.deleteInbound(inbound.id);
    console.log(`✅ Deleted inbound: ${inbound.tag}`);

  } catch (error) {
    if (error instanceof APIError) {
      console.error(`❌ Error: ${error.message} (Status: ${error.statusCode})`);
    } else {
      console.error(`❌ Error: ${error}`);
    }
  }
}

async function main() {
  console.log('LX-Proxy Node.js SDK - Examples\n');
  console.log('='.repeat(50));

  // Run examples
  await exampleBasicUsage();
  await exampleUserManagement();
  await exampleInboundManagement();

  console.log('\n' + '='.repeat(50));
  console.log('✅ All examples completed!\n');
}

// Run if executed directly
if (require.main === module) {
  main().catch(console.error);
}
